src/braket/circuits/noise_model/noise_model.py

```diff
@@ -365,6 +401,51 @@ def from_dict(cls, noise_dict: dict) -> NoiseModel:
         )
         return model
 
+    def _apply_measure_noise(self, circuit: Circuit) -> Circuit:
+        """Apply measure noise to the circuit.
+
+        Args:
+            circuit (Circuit): The circuit to apply measure noise to.
+
+        Returns:
+            Circuit: The circuit with measure noise applied.
+        """
+        result = Circuit()
+        i = 0
+        n = len(circuit.instructions)
+        while i < n:
+            instr = circuit.instructions[i]
+            if isinstance(instr.operator, Measure):
+                # Collect contiguous block of measure instructions
+                measure_block = []
+                while i < n and isinstance(circuit.instructions[i].operator, Measure):
```

Comment:
I don't think the logic makes sense here. If there is `N` instruction in a circuit and `M` noise in a noise model. Your code runs O(N^2*M). Then, why not just iterate through the circuit instructions. And for every instruction, iterate through noise in a noise model. In this way, the complexity is O(N*M).